src/providers/wallet/wallet.ts

```diff
@@ -454,8 +455,15 @@ export class WalletProvider {
     return new Promise((resolve, reject) => {
       this.persistenceProvider
         .getLastAddress(wallet.id)
-        .then(addr => {
-          if (!forceNew && addr) return resolve(addr);
+        .then((addr: string) => {
+          if (addr) {
+            // prevent to show legacy address
+            const isBchLegacy =
+              wallet.coin == 'bch' &&
+              ['C', 'H', 'm', 'n'].indexOf(addr.charAt(0)) > -1;
```

Comment:
this seems a little difficult to read.

maybe:
 const isBchLegacy = 
             wallet.coin == 'bch' &&  addr.match(/^[CHmn]/;

Reply:
not sure...

Reply:
I agree. Done!